<b>DATASET</b>

In [5]:
pip install folium

Note: you may need to restart the kernel to use updated packages.


In [6]:
import numpy as np

import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
import requests
from pandas.io.json import json_normalize
import folium
import matplotlib.pyplot as plt


print(' Successfully imported Libraries.')

 Successfully imported Libraries.


In [7]:
CLIENT_ID = 'TZ4U5TAKOKC51EJALBKLAG1UPWLP2HSABKIQ001Y2WWC25L4' 
CLIENT_SECRET = '5COIL4Q2LO5AAUOJOXL2WONTBASRHZV0Y0RHOKER5ZNEKOPR' 
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)

Your credentails:
CLIENT_ID: TZ4U5TAKOKC51EJALBKLAG1UPWLP2HSABKIQ001Y2WWC25L4


In [8]:
LIMIT = 500 # Maximum is 100
cities = ["Lansing, MI", 'Grand Rapids, MI', 'Traverse City, MI', 'Royal Oak, MI', 'Ann Arbor, MI']
results = {}
for city in cities:
    url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&near={}&limit={}&categoryId={}'.format(
        CLIENT_ID, 
        CLIENT_SECRET, 
        VERSION, 
        city,
        LIMIT,
        "4bf58dd8d48988d1d2941735") # Sushi Restaurant CATEGORY ID
    results[city] = requests.get(url).json()

In [9]:
df_venues={}
for city in cities:
    venues = json_normalize(results[city]['response']['groups'][0]['items'])
    df_venues[city] = venues[['venue.name', 'venue.location.address', 'venue.location.lat', 'venue.location.lng']]
    df_venues[city].columns = ['Name', 'Address', 'Lat', 'Lng']

<b>DATA VISUALIZATION</b>

In [10]:
maps = {}
for city in cities:
    city_lat = np.mean([results[city]['response']['geocode']['geometry']['bounds']['ne']['lat'],
                        results[city]['response']['geocode']['geometry']['bounds']['sw']['lat']])
    city_lng = np.mean([results[city]['response']['geocode']['geometry']['bounds']['ne']['lng'],
                        results[city]['response']['geocode']['geometry']['bounds']['sw']['lng']])
    maps[city] = folium.Map(location=[city_lat, city_lng], zoom_start=12)

    # add markers to map
    for lat, lng, label in zip(df_venues[city]['Lat'], df_venues[city]['Lng'], df_venues[city]['Name']):
        label = folium.Popup(label, parse_html=True)
        folium.CircleMarker(
            [lat, lng],
            radius=5,
            popup=label,
            color='blue',
            fill=True,
            fill_color='#3186cc',
            fill_opacity=0.7,
            parse_html=False).add_to(maps[city])  
    print(f"Total Number of Sushit Restaurants in {city} = ", results[city]['response']['totalResults'])

Total Number of Sushit Restaurants in Lansing, MI =  51
Total Number of Sushit Restaurants in Grand Rapids, MI =  55
Total Number of Sushit Restaurants in Traverse City, MI =  4
Total Number of Sushit Restaurants in Royal Oak, MI =  31
Total Number of Sushit Restaurants in Ann Arbor, MI =  61


In [11]:
maps[cities[0]]

In [13]:
maps[cities[1]]

In [14]:
maps[cities[2]]

In [15]:
maps[cities[3]]

In [16]:
maps[cities[4]]

<b>DATA ANALYSIS</b>

In [18]:
import math

In [20]:
LP=114297
GRP=188040
TCP=14674
ROP=57236
AAP=113934

LR=51
GRR=55
TCR=4
ROR=31
AAR=61

def round_up(n,decimals=7):
        multiplier=10**decimals
        return int(n*multiplier)/multiplier

print('Lansing Sushi Restaurant to Population Lansing:          '+ str((round_up(LR/LP))))
print('Lansing Sushi Restaurant to Population Grand Rapids:     '+ str((round_up(GRR/GRP))))
print('Lansing Sushi Restaurant to Population Traverse City:    '+ str((round_up(TCR/TCP))))
print('Lansing Sushi Restaurant to Population Royal Oak:        '+ str((round_up(ROR/ROP))))
print('Lansing Sushi Restaurant to Population Ann Arbor:        '+ str((round_up(AAR/AAP))))

Lansing Sushi Restaurant to Population Lansing:          0.0004462
Lansing Sushi Restaurant to Population Grand Rapids:     0.0002924
Lansing Sushi Restaurant to Population Traverse City:    0.0002725
Lansing Sushi Restaurant to Population Royal Oak:        0.0005416
Lansing Sushi Restaurant to Population Ann Arbor:        0.0005353


In [21]:
print('Lansing Sushi Restaurant to Population Lansing:          '+ str((round_up(LP/LR))))
print('Lansing Sushi Restaurant to Population Grand Rapids:     '+ str((round_up(GRP/GRR))))
print('Lansing Sushi Restaurant to Population Traverse City:    '+ str((round_up(TCP/TCR))))
print('Lansing Sushi Restaurant to Population Royal Oak:        '+ str((round_up(ROP/ROR))))
print('Lansing Sushi Restaurant to Population Ann Arbor:        '+ str((round_up(AAP/AAR))))

Lansing Sushi Restaurant to Population Lansing:          2241.117647
Lansing Sushi Restaurant to Population Grand Rapids:     3418.9090909
Lansing Sushi Restaurant to Population Traverse City:    3668.5
Lansing Sushi Restaurant to Population Royal Oak:        1846.3225806
Lansing Sushi Restaurant to Population Ann Arbor:        1867.7704918


<b>RESULTS</b>

<i>
    Looking at the data for each city we can digest the following for each sushi restaurant there is the potential for X number of customers per city:

Lansing = 2241 Grand Rapids = 3418 Traverse City=3668 Royal Oak=1846 Ann Arbor=1867

Reviewing the data it would be the client's best option to open a starter franchise in either Traverse City or Grand Rapids because the compeition is relatively the same. One factor to keep in mind is that the Asian population according tot he 2010 Census shows that only 102 individuals living in Traverse City is of Asian decent whereas in Grand Rapdids there are 4323 individuals of Asian decent.
</i>
    